packages/amplication-build-manager/src/build-runner/build-runner.service.ts

```diff
@@ -31,6 +32,40 @@ export class BuildRunnerService {
     private readonly logger: AmplicationLogger
   ) {}
 
+  async onPackageManagerCreateResponse(buildId: string) {
+    await this.dsgCompleted(buildId);
+  }
+
+  async runPackageGenerator(jobBuildId: string, resourceId: string) {
+    const buildId = this.buildJobsHandlerService.extractBuildId(jobBuildId);
+
+    const dsgResourceData =
+      await this.buildJobsHandlerService.extractDsgResourceData(jobBuildId);
```

Comment:
you are calling extractDsgResourceData twice - maybe you can send the packages from the parent